tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -229,38 +297,65 @@ def get_asts(self) -> List[chapel.core.AstNode]:
         with self.context.track_errors() as _:
             return self.parse_file()
 
-    def rebuild_segments(self):
+    def rebuild_index(self):
         """
         Rebuild the cached line info and siblings information
 
         Note: this is a potentially expensive operation, it should only be done
         when advancing the revision
         """
         asts = self.parse_file()
```

Comment:
This function has gotten long and unwieldy, can you break it up with some comments and/or separate into "private" `_` methods that are documented

Reply:
I've split into some helpers and made use of visitor methods to clean it up.